test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -297,7 +297,19 @@ object Assertion extends AssertionVariants {
    * Makes a new assertion that requires an exception to have a certain message.
    */
   def hasMessage(message: String): Assertion[Throwable] =
-    assertion[Throwable]("hasMessage")(param(message))(th => th.getMessage == message)
+    hasMessage(equalTo(message))
+
+  /**
+   * Makes a new assertion that requires an exception to have a certain message.
+   */
+  def hasMessage(message: Assertion[String]): Assertion[Throwable] =
+    Assertion.assertionRec("hasMessage")(param(message))(message)(th => Some(th.getMessage))
+
+  /**
+   * Makes a new assertion that requires an exception to have a certain cause.
+   */
+  def hasCause(cause: Assertion[Throwable]): Assertion[Throwable] =
```

Comment:
As more assertions are added to `Assertion` object maybe we should start name spacing them by the type of object the are asserting on. 
For example we could have a nested object `throwable` and put `hasCause` and  `hasMessage` under it.

Reply:
That's an interesting idea. We could definitely go in that direction. The downside is it makes it a little more verbose to write everything because you need one more qualifier but you're right if we have a ton more assertions it could provide a way to organize it. I think I would prefer to try to stay with the flat structure for now and revisit down the road if we need to.

Reply:
removing `Assetion.hasMessage(message: String)`